sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryStrategy.java

```diff
@@ -57,6 +60,12 @@ default boolean shouldRetry(HttpResponse httpResponse) {
      * @return Whether a retry should be attempted.
      */
     default boolean shouldRetryException(Throwable throwable) {
-        return throwable instanceof Exception;
+        if ((throwable instanceof HttpResponseException)) {
+            int statusCode = (((HttpResponseException) throwable).getResponse().getStatusCode());
+
+            return statusCode != 401 && statusCode != 403;
+        }
+
+        return throwable instanceof IOException || throwable instanceof AzureException;
```

Comment:
have we tried mapping `MsalException` to some core exception type?
The change in `RetryStrategy` is into right direction but isn't this masking underlying problem? I.e. should `MsalExeptions` propagate from identity unwrapped?

As for the `RetryStrategy` change:
- should we exclude all 4xx errors?
- is it safe to retry any `AzureException` ?